Sources/StreamChat/Models/Channel.swift

```diff
@@ -164,7 +165,7 @@ public struct _ChatChannel<ExtraData: ExtraDataTypes> {
         currentlyTypingMembers: Set<_ChatChannelMember<ExtraData.User>> = [],
         lastActiveWatchers: @escaping (() -> [_ChatUser<ExtraData.User>]) = { [] },
         team: TeamId? = nil,
-        unreadCount: ChannelUnreadCount = .noUnread,
+        unreadCount: @escaping () -> ChannelUnreadCount = { .noUnread },
```

Comment:
so `unreadCount` is not a value but a dynamic thingy now.. This makes us move further away from "snapshot models". Since we send `channelUpdated` delegate callbacks when unread count updates, why don't we still use static unread counts in models?

Reply:
Nope, this just means it's computed lazily. Getting unread count requires iterating over channel reads + an additional DB query. That's why it's computed lazily. See other `@Cached` properties in our models.

Reply:
OK, explanation + re-reading code made it clearer. Makes sense 👌